Jenkinsfile.windows

```diff
@@ -0,0 +1,281 @@
+                            def packageName = "RStudio-${PACKAGE_VERSION}"
+
+                            // strip unhelpful suffixes from filenames
+                            bat "move package\\win32\\build\\${packageName}-RelWithDebInfo.exe package\\win32\\build\\${packageName}.exe"
+                            bat "move package\\win32\\build\\${packageName}-RelWithDebInfo.zip package\\win32\\build\\${packageName}.zip"
+
+                            // windows docker container cannot reach instance-metadata endpoint. supply credentials at upload.
+                            withAWS(role: 'jenkins', roleAccount: AWS_ACCOUNT_ID) {
+                              retry(5) {
+                                bat "aws s3 cp package\\win32\\build\\${packageName}.exe ${buildDest}/${packageName}.exe"
+                                bat "aws s3 cp package\\win32\\build\\${packageName}.zip ${buildDest}/${packageName}.zip"
+                              }
+                            }
+                          }
+                        }
+                      }
+
+                      stage('Sentry Upload') {
+                        environment {
+                          SENTRY_API_KEY = credentials('ide-sentry-api-key')
+                        }
+
+                        steps {
+                          retry(5) {
+                            // convert the PDB symbols to breakpad format (PDB not supported by Sentry)
+                            timeout(activity: true, time: 15) {
+                              bat '''
+                                cd package\\win32\\build
+                                FOR /F %%G IN ('dir /s /b *.pdb') DO (..\\..\\..\\dependencies\\windows\\breakpad-tools-windows\\dump_syms %%G > %%G.sym)
+                              '''
+
+                              // upload the breakpad symbols, attempt to run sentry uplaod
+                              bat "cd package\\win32\\build\\src\\cpp && ..\\..\\..\\..\\..\\dependencies\\windows\\sentry-cli.exe --auth-token %SENTRY_API_KEY% upload-dif --log-level=debug --org rstudio --project ide-backend -t breakpad ."
+                            }
+                          }
+                        }
+                      }
+
+                      stage ('Publish') {
+                        environment {
+                          GITHUB_LOGIN = credentials('github-rstudio-jenkins')
+                        }
+
+                        steps {
+                          script {
+                            def packageName = "RStudio-${PACKAGE_NAME}"
+
+                            // derive product
+                            def product = "${FLAVOR}"
+                              if (IS_PRO) {
+                                product = "${FLAVOR}-pro"
+                              }
+
+                            // publish the build (self installing exe)
```

Comment:
Currently the `rhel8 electron` build will also perform URL redirects for windows installers, because there's no `publish-daily-binary.bat` equivalent. See [current Jenkinsfile](https://github.com/rstudio/rstudio-pro/blob/35fd2a914cd7d914ae535e39be8baa4cbf00ae68/Jenkinsfile#L131-L135)

We should either
1.  Write that .bat file so we can update the URL in windows (potentially takes more dev time)
2. Add a stage at the end of this pipeline that will pull  (just pull, not pullbuildpush) one of the linux docker images and run the `publish-daily-binary.sh` for the windows binary (longer build time)

Reply:
Writing an entire batch script for this seems like a major time sink  at this point. We'll have to see how much longer it actually takes to switch back over to a docker agent and run this script, but it's probably less than the cost of writing and maintaining two scripts that do the same thing.

Reply:
I can't say how long switching agents will take, but I just did a custom dispatch test to make sure that the checkout stage works and it's [quick](https://build.posit.it/blue/organizations/jenkins/IDE%2Fexperiments%2Fmacos-matrix/detail/macos-matrix/28/pipeline/31), at least for the macOS pipeline. Running `rstudio-version.sh` is also pretty much instantaneous now too.